src/validation.cpp

```diff
@@ -2402,6 +2444,10 @@ static bool ActivateBestChainStep(CValidationState& state, const CChainParams& c
         }
     }
 
+    // If any blocks were disconnected, disconnectpool may be non empty.  Add
+    // any disconnected transactions back to the mempool.
+    ReacceptToMemoryPool(disconnectpool);
```

Comment:
nit: Any reason to put `ReacceptToMemoryPool` here and not under `if (fBlocksDisconnected)`?
It might make the logic slightly clearer.